docs/categorical-histogram-dap-04.md

```diff
@@ -0,0 +1,113 @@
+---
+slug: /categorical-histogram-dap-04
+sidebar_position: 2
+---
+
+# Emulating Categorical Histograms with DAP draft-04
+
+The Prio3Histogram VDAF has changed between drafts 04 and 07 of DAP. Initially,
+it provided the functionality of a conventional histogram of a continuous
+variable. As of draft-07, its API has changed to be more flexible, supporting
+other use cases like categorical variables, multi-dimensional histograms, etc.
+This guide covers how to emulate this behavior and use Prio3Histogram as of DAP
+draft-04 to count occurrences of different values of a categorical variable.
+This is only applicable to deployments using older versions of the protocol;
+newer deployments using draft-07 or later can disregard this guide.
+
+## Background
+
+### Prio3Histogram, DAP draft-04
+
+This version uses the [definition of Prio3Histogram from VDAF
+draft-05](https://www.ietf.org/archive/id/draft-irtf-cfrg-vdaf-05.html#name-prio3histogram).
+The VDAF has one parameter, `buckets`, which is a list of integers in ascending
+order. The measurement is an integer, which gets compared to each value in the
+`buckets` parameter.
+
+Measurements are encoded into a vector that has a 1 in one element and a 0
+elsewhere. The length of the encoded vector is one more than the length of the
+`buckets` parameter. The encoding procedure divides the number line into
+intervals, separated at the numbers in the `buckets` parameter, and assigns each
+element of the encoded measurement vector to one of the intervals, in increasing
+order. Then, for any measurement, the interval that contains that measurement
+value is identified, the corresponding vector element is set to 1, and the rest
+are set to 0. Each interval includes its maximum boundary value and excludes its
+minimum boundary value. In particular, if the measurement is less than or equal
+to `buckets[0]`, then the output is `<1, 0, 0, 0, ... 0>`. If the measurement is
+greater than `buckets[0]` and less than or equal to `buckets[1]`, then the
+output is `<0, 1, 0, 0, ... 0>`. If the measurement is greater than
+`buckets[len(buckets) - 1]`, then the output is `<0, 0, 0, ... 0, 1>`.
+
+### Prio3Histogram, DAP draft-07
+
+This version uses the [definition of Prio3Histogram from VDAF
+draft-07](https://www.ietf.org/archive/id/draft-irtf-cfrg-vdaf-07.html#name-prio3histogram).
+The vdaf has two parameters, `length` and `chunk_length`, both positive
+integers. The measurement is an integer between 0 and (length - 1).
```

Comment:
in other places, instead of parenthesizing expressions like `number - 1`, we use backticks / code blocks